### PR TITLE
Multilabel randomForestSRC

### PR DESCRIPTION
**Pull request by PhilippPro**
new branch for multilabel randomForestSRC.

Followed recommendations of #684.

### CODE REVIEW

**Comment by larskotthoff** [comment]
Is this ready to be merged?

Please use `stri_paste` instead of `paste`. Your test doesn't seem to check anything?

**Comment by PhilippPro** [comment]
Yes there is missing something. I just copied the test from https://github.com/mlr-org/mlr/blob/master/tests/testthat/test_classif_randomForestSRC.R.
Can I just apply `testSimpleParset` analogously to the randomForestSRC case (see below)? 
I do not know where this function comes from... 

```
  testSimpleParsets("classif.randomForestSRC", binaryclass.df, binaryclass.target, binaryclass.train.inds, old.predicts.list, parset.list)
  testProbParsets ("classif.randomForestSRC", binaryclass.df, binaryclass.target, binaryclass.train.inds, old.probs.list, parset.list)
```

**Comment by larskotthoff** [comment]
The functions are defined in helper_helpers.R. You should be able to just use them.

**Comment by berndbischl** [comment]
> f = as.formula(paste("cbind(", paste(targets, collapse = ",", sep = " "), ")  ~ .", sep = ""))

I dont like this. Can you please help to extend getTaskFormula so such a formula is created?
That function already supports "multilabel" but i guess you need an extra option to include a "cbind" here for the left hand side, where the targets are.

**Comment by berndbischl** [comment]
```
+  d = getTaskData(.task, .subset)
 +  for(i in targets)
 +    d[i] = factor(d[[i]], levels = c("TRUE", "FALSE"))
```

I also dislike this here. we probably want to have an option in getTaskData to do that conversion there.

**Comment by PhilippPro** [comment]
@berndbischl 
1. Do you really think we need a cbind option in getTaskFormula? This is a very special case, isn't it?
2. Maybe the same for getTaskData

Can somebody tell me, what is doing jenkins in comparison to travis? (and how I can login to jenkins)

**Comment by schiffner** [comment]
Two mini things about the learner params:
* For consistency the `forest` parameter should also be included (see @MariaErdmann 's PR #992 )
* In the newest version of randomForestSRC observation weights are supported.

**Comment by larskotthoff** [comment]
@PhilippPro Email me to get a Jenkins account.

**Comment by berndbischl** [comment]
> 1. Do you really think we need a cbind option in getTaskFormula? This is a very special case, isn't it?
> 2. Maybe the same for getTaskData

for 1) i might agree. for 2) this seems pretty "expectable" that some methods would like to have the target as factors?


> In the newest version of randomForestSRC observation weights are supported.

yup, see "case.wt". so you need to set the property and pass .weights to this.

**Comment by berndbischl** [comment]
@PhilippPro 
Philipp:

```
  if (type == "multilabel_factor")
    return(lapply(y, function(x) factor(x, levels = c("TRUE", "FALSE"))))
```

You have performed a change to one of mlr's most important core functions:
- you did not document this
- all other options for recode.target used a "dot" as word separator, you now used a "_". please call it 
- you did not unit test this function itself

please always thing about stuff like this when you do API changes

**Comment by PhilippPro** [comment]
All the others are "drop.levels" ?;). 

I just did this based on your recommendation, so I thought this is obvious, that I will change something there. 
Where are the tests for TaskOperators?

**Comment by berndbischl** [comment]
> All the others are "drop.levels" ?;).

yes, but also true for other functions in mlr! (its basically the style guide)

> I just did this based on your recommendation, so I thought this is obvious, that I will change something there.

this sentence does not "parse" for me. do you mean: 
"my PR was not  done, it was just a first try"
in general, please communicate a bit after more commits to the PR so we know whats going on.
either say: "ok did some stuff, but note done" or "hey, i guess this is OK from my side, pls review!"

> Where are the tests for TaskOperators?

i guess test_base_getTaskData is the best place!

thx!

**Comment by PhilippPro** [comment]
Ok, I think it is ready to be merged (although Bernd will surely find some mistakes ;)). 
Sorry for the amount of commits, I sometimes just see the mistakes after committing.

**Comment by berndbischl** [comment]
> Sorry for the amount of commits, I sometimes just see the mistakes after committing.

commits can now be squashed from our side very easily on merge, so i dont care if you 100 of those, where in each one you convert a single underscore in a dot :)

**Comment by MasonGallo** [comment]
@PhilippPro I sometimes do this too...I edited the wiki to @berndbischl's point so hopefully it's clear now 👍

**Comment by berndbischl** [comment]
> @PhilippPro I sometimes do this too...I edited the wiki to @berndbischl's point so hopefully it's clear now

thx!

File: R/Task_operators.R [hunk 1]
```diff
@@ -323,6 +323,8 @@ recodeY = function(y, type, td) {
     return(as.numeric(2L * (y == td$positive) - 1L))
   if (type %in% c("lcens", "rcens", "icens"))
     return(recodeSurvivalTimes(y, from = td$censoring, to = type))
+  if (type == "multilabel.factor")
```
Comment (schiffner) [inline R/Task_operators.R]: Ideally, this option should also be documented (`recode.target` param of `getTaskData`, l. 229`)

**Comment by berndbischl** [comment]
1) pls doc the API change as Julia (and I before...) requested

2) expect_true(all(sapply(df[, multilabel.target], is.factor)))

this is really not enough for a proper test. you also need to check that the conversion has the correct contents.  this is most simply if you create a mini-task with very few rows and cols for the test...

3) multilabel.formula = as.formula(paste("cbind(", paste(multilabel.target, collapse = ",", sep = " "), ")  ~ .", sep = ""))
 
*YOU* were the one who told me that this formula structure does LIKELY ONLY holds for this learner. that was your argument not to change the getTaskFormula getter. Now you put it in a place where it would need to hold for many other learners to be of use.....?
At least call it maybe multilabel.formula.cbind

**Comment by PhilippPro** [comment]
In my opinion it is mergeable now.

**Pull request merged**